api/repositories/service_binding_repository.go

```diff
@@ -142,3 +152,57 @@ func cfServiceBindingToRecord(binding v1alpha1.CFServiceBinding) ServiceBindingR
 		},
 	}
 }
+
+func (r *ServiceBindingRepo) ListServiceBindings(ctx context.Context, authInfo authorization.Info, message ListServiceBindingsMessage) ([]ServiceBindingRecord, error) {
+	nsList, err := r.namespacePermissions.GetAuthorizedSpaceNamespaces(ctx, authInfo)
+	if err != nil {
+		// untested
+		return nil, fmt.Errorf("failed to list namespaces for spaces with user role bindings: %w", err)
+	}
+
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		// untested
+		return []ServiceBindingRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	var filteredServiceBindings []servicesv1alpha1.CFServiceBinding
+	for ns := range nsList {
+		serviceInstanceList := new(servicesv1alpha1.CFServiceBindingList)
+		err = userClient.List(ctx, serviceInstanceList, client.InNamespace(ns))
+		if k8serrors.IsForbidden(err) {
+			continue
+		}
+		if err != nil {
+			// untested
+			return []ServiceBindingRecord{}, fmt.Errorf("failed to list service instances in namespace %s: %w", ns, err)
+		}
+		filteredServiceBindings = append(filteredServiceBindings, applyServiceBindingListFilter(serviceInstanceList.Items, message)...)
+	}
+
+	return returnServiceBindingList(filteredServiceBindings), nil
```

Comment:
`return returnXXX` sounds weird, maybe change the name to describe what the function actually does, e.g. `toServiceBindingRecords`